components/tree-grid/index.jsx

```diff
@@ -0,0 +1,198 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// Implements the [Welcome Mat design pattern](https://lightningdesignsystem.com/components/welcome-mat/) in React.
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+// ### shortid
+// [npmjs.com/package/shortid](https://www.npmjs.com/package/shortid)
+// shortid is a short, non-sequential, url-friendly, unique id generator
+import shortid from 'shortid';
+
+import Row from './private/row';
+import Icon from '../icon';
+import Button from '../button';
+
+import { TREE_GRID } from '../../utilities/constants';
+
+const displayName = TREE_GRID;
+
+const propTypes = {
+	/**
+	 * CSS class names to be added to the container element. `array`, `object`, or `string` are accepted.
+	 */
+	className: PropTypes.oneOfType([
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	/**
+	 * HTML id for component.
+	 */
+	id: PropTypes.string,
+
+	labels: PropTypes.shape({
+		chooseRow: PropTypes.string,
+		selectAll: PropTypes.string,
+	}),
+
+	data: PropTypes.object.isRequired,
+
+	isSingleSelect: PropTypes.bool,
+};
+
+/**
+ * A tree is visualization of a structure hierarchy. A branch can be expanded or collapsed.
+ */
+class TreeGrid extends React.Component {
+	componentWillMount() {
+		this.generatedId = shortid.generate();
+	}
+
+	/**
+	 * Get the TreeGrid's HTML id. Generate a new one if no ID present.
+	 */
+	getId() {
+		return this.props.id || this.generatedId;
+	}
+
+	render() {
+		return (
+			<table
+				id={this.getId()}
+				aria-multiselectable="true"
+				className={classNames(
+					'slds-table',
+					'slds-table_bordered',
+					'slds-table_edit',
+					'slds-table_fixed-layout',
+					'slds-table_resizable-cols',
```

Comment:
Unless you are implementing resizable, this might need to be removed. I'd recommend not implementing it. It's not in the current Data either. Column width on the other hand is needed.